standard/StandardProject.cmake

```diff
@@ -0,0 +1,122 @@
+#[[
+Simple project setup. There's a fair amount of boiler plate that has to happen
+in a particular order around project setup. This function wraps project() and
+does the following:
+
+* Changes CMake around so that the output variables follow the /bin and /lib output
+  directory convention popular on Gnu.
+* Ensures that ARM and 64-bit code are built as position-independent code
+* Verifies that the compiler actually supports C++11
+* Sets the correct flags to enable C++11 on all platforms
+* Ensures compilation happens with visibility set to hidden by default
+* Builds fat binaries on mac
+* Sets standard_BUILD* variables for consumption later down the line.
+* Enforces the project has a VERSION set.
+]]
+function(standard_project project_name)
```

Comment:
I know you need to do things before _and_ after the `project` command, but I prefer being able to see the `project` command explicitly in my project.  Can you break this into a `standard_before_project` and `standard_after_project` command pair instead?